### ISSUE TITLE
rebar3 not creating numerl.so

### ISSUE THREAD

**Issue by dendrown** [tanguyl/numerl#15]
Hello,

I am attempting to try out numerl, but I am having an issue.  Although the build appears to be successful, there is no numerl.so created.  I have tried both with my own project and with the raytracer demo on github.  I have openblas and lapack installed, and I have tried using erlang 22, 23, and 24, but no luck.
```bash
$ rebar3 compile
===> Verifying dependencies...
===> Fetching numerl (from {git,"https://github.com/tanguyl/numerl.git",
                  {ref,"bf10686858d1c931c7bd6ab38ed9ddae49bf1b04"}})
===> Fetching pc v1.14.0
===> Analyzing applications...
===> Compiling pc
===> Compiling c_src/numerl.c
===> Linking /tmp/raytracer/_build/default/lib/numerl/priv/numerl.so

$ ls -al /tmp/raytracer/_build/default/lib/numerl/priv/
total 2
drwxr-xr-x 2 dend dend 1024 May  3 14:11 .
drwxr-xr-x 7 dend dend 1024 May  3 14:11 .. 
```

Am I missing something?

Thanks,
den

**Comment by tanguyl** [comment]
Hi Den;
I'm sorry I answer this late. This is a bit peculiar; I'm not sure what can cause this issue. So far, I've been sticking to Erlang 22, and I'm fairly certain I tested on 23 and 24. I do have a handfull of questions for you:

- Which OS are you using?
- How did you install blas-lapacke?
- Is your project a clean "git clone https://github.com/tanguyl/raytracer.git" ?

Hoping to hear back from you,
Tanguy

**Comment by dendrown** [comment]
Hi Tanguy,

I am using Gentoo Linux.  The dependencies are installed with Portage's emerge.  Portage is the standard package manager for Gentoo.  On my machine with Erlang 22:

```bash
equery l erlang openblas lapack
 * Searching for erlang ...
[I--] [??] dev-lang/erlang-22.3.2:0/22.3.2

 * Searching for openblas ...
[I--] [??] sci-libs/openblas-0.3.9-r1:0

 * Searching for lapack ...
[IP-] [  ] sci-libs/lapack-3.8.0-r1:0
```

The "??" simply means the installed version of the package is no longer available on the Portage tree.  I don't have that on the more up-to-date systems.

And yes, I was testing using a fresh install of the sample program:
```bash
$ git clone https://github.com/tanguyl/raytracer.git 
Cloning into 'raytracer'...
remote: Enumerating objects: 63, done.
remote: Counting objects: 100% (63/63), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 63 (delta 23), reused 43 (delta 9), pack-reused 0
Receiving objects: 100% (63/63), 329.99 KiB | 2.46 MiB/s, done.
Resolving deltas: 100% (23/23), done.

$ cd raytracer 
$ rebar3 compile
===> Verifying dependencies...
===> Fetching numerl (from {git,"https://github.com/tanguyl/numerl.git",
                  {ref,"bf10686858d1c931c7bd6ab38ed9ddae49bf1b04"}})
===> Fetching pc v1.14.0
===> Analyzing applications...
===> Compiling pc
===> Compiling c_src/numerl.c
===> Linking /tmp/raytracer/_build/default/lib/numerl/priv/numerl.so
$ ls -l _build/default/lib/numerl/priv 
total 0
```

I also see that there are no beam files for raytracer.erl, nor numerl.erl.  It's like the build fails in the midway, but gives no indication that it failed.

```bash
$ find . -name \*.beam
./_build/default/plugins/pc/ebin/pc_port_env.beam
./_build/default/plugins/pc/ebin/pc_compilation.beam
./_build/default/plugins/pc/ebin/pc_util.beam
./_build/default/plugins/pc/ebin/pc_port_specs.beam
./_build/default/plugins/pc/ebin/pc.beam
./_build/default/plugins/pc/ebin/pc_prv_compile.beam
./_build/default/plugins/pc/ebin/pc_prv_clean.beam
```

Thanks for looking at this.

Cheers,
den

**Comment by tanguyl** [comment]
Thanks for your answer! I'll try to install numerl on a gentoo distro this week.

**Comment by dendrown** [comment]
Thank you! :)

**Comment by dendrown** [comment]
Hello again.  I'm guessing you had some trouble installing Gentoo.  It is certainly not a quick-n-easy procedure like most Linux distributions.  Anyway, I determined that I am able to create `numerl.so` if I change this line in `rebar.config`:
```erlang
  {"linux", "LDFLAGS", "$LDFLAGS -lopenblas -llapacke -lm"},
```
to this:
```erlang
  {"linux", "LDFLAGS", "$LDFLAGS -lopenblas -llapack -lm"},
```
(Note the final 'e' missing for liblapack.so.)

The build succeeds as do your eunit tests.  The question I have now is how to get the rebar3 port compiler to differentiate between Gentoo and other Linux distributions.

**Comment by dendrown** [comment]
Okay, so I have found the problem.  It was an issue with the dependencies on my system.  I had installed the base lapack implementation without installing the virtual lapacke package for the interface.
```bash
root@phoebe:/etc/portage# equery l lapack\*
 * Searching for lapack* ...
[IP-] [  ] sci-libs/lapack-3.10.0:0
[IP-] [  ] virtual/lapacke-3.8-r1:0
```
I can now successfully run the test suite as well as your [raytracer demo](https://github.com/tanguyl/raytracer.git) without making any changes to the numerl's `rebar.config`.
